File: src/server/src/main/java/com/project/domain/workout/WorkoutHistory.java
package com.project.domain.workout;

import com.project.domain.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import java.util.UUID;

@Entity
public class WorkoutHistory extends BaseEntity {
    @Id
    private final UUID id;

    private UUID workoutId;

    protected WorkoutHistory() {
        super();
        this.id = UUID.randomUUID();
    }

    public WorkoutHistory(UUID workoutId) {
        super();
        this.id = UUID.randomUUID();
        this.workoutId = workoutId;
    }

    public UUID getId() {
        return this.id;
    }

    public UUID getWorkoutId() {
        return this.workoutId;
    }
}

File: src/server/src/main/kotlin/com/project/domain/workout/WorkoutHistory.kt
package com.project.domain.workout

import com.project.common.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity
class WorkoutHistory protected constructor(): BaseEntity() {
    @Id
    val id: UUID = UUID.randomUUID()

    var workoutId: UUID? = null
        private set

    constructor(workoutId: UUID) : this() {
        this.workoutId = workoutId
    }
}
